target/linux/ath79/dts/qca9558_tl-wr1043nd.dtsi

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system {
+			label = "tp-link:green:system";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+
+		usb {
+			label = "tp-link:green:usb";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wifi_green {
+			label = "tp-link:green:wlan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wifi_wps {
+			label = "tp-link:green:wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		button@0 {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		button@1 {
+			label = "RFKILL button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+/* no device connected
+&pcie0 {
+	status = "okay";
+};
+*/
+
+&uart {
+	status = "okay";
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+};
+
+/* missing connector
+&usb1 {
+	status = "okay";
+};
+*/
+
+&pll {
+	clocks = <&extosc>;
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot:	partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
+				label = "firmware";
+				reg = <0x020000 0x7d0000>;
+			};
+
+			partition@7f0000 {
+				label = "art";
+				reg = <0x7f0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		 qca,ar8327-initvals = <
+			0x04 0x00080080 /* PORT0 PAD MODE CTRL */
+			0x0c 0x07600000 /* PORT6 PAD MODE CTRL */
+			0x50 0xcf35cf35 /* LED_CTRL0 */
+			0x54 0xcf35cf35 /* LED_CTRL1 */
+			0x7c 0x0000007e /* PORT0_STATUS */
+			0x94 0x0000007e /* PORT6 STATUS */
+			>;
```

Comment:
I would prefer the same values as in mach file...
```
qca,ar8327-initvals = <
    0x04 0x00080080 /* PORT0 PAD MODE CTRL */
    0x0c 0x07600000 /* PORT6 PAD MODE CTRL */
    0x10 0x81000080 /* POWER_ON_STRIP */
    0x50 0xcc35cc35 /* LED_CTRL0 */
    0x54 0xca35ca35 /* LED_CTRL1 */
    0x58 0xc935c935 /* LED_CTRL2 */
    0x5c 0x03ffff00 /* LED_CTRL3 */
    0x7c 0x0000007e /* PORT0_STATUS */
    0x94 0x0000007e /* PORT6 STATUS */
    >;
```
I have tested this on my wr1043v3 and works fine.

Reply:
I'm just for the minimum options, it works fine without them, should we add useless options ? do you think they matter ?

Reply:
To this, i would prefer the full values because the defaults of some registers are not the same values as i had seen in the mach file... For example: LED_CTRL3 defaults to 0x03ffffff instead of 0x03ffff00 and POWER_ON_STRIP to 0x0000002a instead of 0x81000080.
LED_CTRL0, LED_CTRL1 and LED_CTRL2 are the same as the default register values...

Buti dont know anything about the side effects if this values not set properly...
For me, i would stick to the values from the mach file...